### PR TITLE
Fix: simultaneous use of gradient_accumulation_steps and max_train_epochs

### PR DESCRIPTION
**Pull request by tsukimiya**
gradient_accumulation_steps を 2以上に指定した場合にmax_train_epochsが正しく設定されない問題の修正です。

現在、gradient_accumulation_stepsを指定するとmax_train_epochsに到達するためのstep数が (gradient_accumulation_steps)倍となってしまい、指定したepoch数で学習が終わらない状態です。

### CODE REVIEW

**Comment by Tomorrowday** [comment]
Has anyone tried training with parameters where 'gradient accumulation*batch size = number of images' in after this fix? I tried it, and the lora turned out to be one that doesn't make any changes to the generation.
